File: Atributos de validación Data Annotations multiidioma/[C#]-Atributos de validación Data Annotations multiidioma/C#/LocalizeDataAnnotations/Infrastructure/TextosBBDD.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Threading;
using System.Web;

namespace LocalizeDataAnnotations.Infrastructure
{
    public static class TextosBBDD
    {

        private static DataTable _textos;

        public static string Recuperar(string textoID)
        {
            if (_textos == null)
            {
                _textos = new DataTable();
                _textos.ReadXml(HttpContext.Current.Server.MapPath("~/Content/Textos.xml"));
            }
            string idioma = Thread.CurrentThread.CurrentUICulture.TwoLetterISOLanguageName;
            if (idioma != "en") idioma = "es";
            DataRow row = _textos.Rows.Find(new object[] { textoID, idioma });
            if (row != null)
                return row["Texto"].ToString();
            else
                return null;
        }

    }
}

File: Atributos de validación Data Annotations multiidioma/[VB]-Atributos de validación Data Annotations multiidioma/VB/LocalizeDataAnnotations/Infrastructure/TextosBBDD.vb
Imports System.Threading

Public NotInheritable Class TextosBBDD

    Private Shared _textos As DataTable

    Public Shared Function Recuperar(textoID As String) As String
        If _textos Is Nothing Then
            _textos = New DataTable()
            _textos.ReadXml(HttpContext.Current.Server.MapPath("~/Content/Textos.xml"))
        End If
        Dim idioma As String = Thread.CurrentThread.CurrentCulture.TwoLetterISOLanguageName
        If idioma <> "en" Then idioma = "es"
        Dim row As DataRow = _textos.Rows.Find(New Object() {textoID, idioma})
        If row IsNot Nothing Then
            Return row("Texto").ToString()
        Else
            Return Nothing
        End If
    End Function

End Class
